### PR TITLE
Add timeout support

### PR DESCRIPTION
**Pull request by jaceksokol**
This is different approach to timeouts (RX based). There are some cases where errors are not propagate through RX extension and the connection is not correctly released. I believe it would require some fundamental refactoring.

### CODE REVIEW

File: src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java [hunk 1]
```diff
@@ -119,6 +124,21 @@ public NettyPgProtocolStream(EventLoopGroup group, SocketAddress address, boolea
...
     }
 
+    private void scheduleSubscriptionCleaner() {
+        Runnable action = () ->
+                Optional.ofNullable(subscribers.peek())
+                        .filter(Subscriber::isUnsubscribed)
+                        .ifPresent(s -> {
+                            ctx.fireExceptionCaught(new IllegalStateException());
+                            ctx.close();
+                        });
```
Comment (jaroslawk) [inline src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java]: this exception will break correctly working communication which is taking more time than 1s.

File: src/main/java/com/github/pgasync/impl/PgConnectionPool.java [hunk 2]
```diff
@@ -209,25 +215,50 @@ public void release(Connection connection) {
...
+        Optional.ofNullable(next)
+                .ifPresent(s -> {
+                    s.onNext(connection);
+                    s.onCompleted();
+                });
+    }
+
+    private void houseKeepSubscribers() {
+        while (!subscribers.isEmpty() && subscribers.peek().isUnsubscribed())
+            subscribers.remove();
```
Comment (jakubiec) [inline src/main/java/com/github/pgasync/impl/PgConnectionPool.java]: shouldn't the peeked subscriber be also unsubscribed?

File: src/main/java/com/github/pgasync/impl/PgConnectionPool.java [hunk 3]
```diff
@@ -142,9 +143,10 @@ public void close() throws Exception {
 
     @Override
     public Observable<Connection> getConnection() {
-        return Observable.<Connection>create(subscriber -> {
+        return Observable.<Connection>unsafeCreate(subscriber -> {
```
Comment (jakubiec) [inline src/main/java/com/github/pgasync/impl/PgConnectionPool.java]: when creating Observables manually the implementer should take care of unsubscription which I cannot find in this class

File: src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java (hunk 1 above)
Comment (jaceksokol) [reply src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java]: If it's a timeout due to long running query than it maybe worth to close connection.

File: src/main/java/com/github/pgasync/impl/PgConnectionPool.java (hunk 3 above)
Comment (jaceksokol) [reply src/main/java/com/github/pgasync/impl/PgConnectionPool.java]: I know, that's why it needs deeper refactoring. I've just changed method name to indicate this.

**Pull request closed without merging**